solr/core/src/java/org/apache/solr/schema/NVector.java

```diff
@@ -0,0 +1,263 @@
+import java.util.Map;
+
+public class NVector extends CoordinateFieldType {
+
+    @Override
+    protected void init(IndexSchema schema, Map<String, String> args) {
+        super.init(schema, args);
+        dimension = 3;
+        createSuffixCache(3);
+    }
+
+    @Override
+    public List<IndexableField> createFields(SchemaField field, Object value) {
+        String externalVal = value.toString();
+        String[] point = parseCommaSeparatedList(externalVal, dimension);
+        String[] nvector = NVectorUtil.latLongToNVector(point);
+
+        List<IndexableField> f = new ArrayList<>((dimension * 2) + 1);
+
+        if (field.indexed()) {
+            for (int i = 0; i < dimension; i++) {
+                SchemaField sf = subField(field, i, schema);
+                f.addAll(sf.createFields(nvector[i]));
+            }
+        }
+
+        if (field.stored()) {
+            f.add(createField(field.getName(), externalVal, StoredField.TYPE));
+        }
+        return f;
+    }
+
+    @Override
+    public ValueSource getValueSource(SchemaField field, QParser parser) {
+        ArrayList<ValueSource> vs = new ArrayList<>(dimension);
+        for (int i = 0; i < dimension; i++) {
+            SchemaField sub = subField(field, i, schema);
+            vs.add(sub.getType()
+                .getValueSource(sub, parser));
+        }
+        return new NVectorValueSource(vs);
+    }
+
+
+    /**
+     * Given a string containing <i>dimension</i> values encoded in it, separated by commas,
+     * return a String array of length <i>dimension</i> containing the values.
+     *
+     * @param externalVal The value to parse
+     * @param dimension   The expected number of values for the point
+     * @return An array of the values that make up the point (aka vector)
+     * @throws SolrException if the dimension specified does not match the number found
+     */
+    public static String[] parseCommaSeparatedList(String externalVal, int dimension) throws SolrException {
+        //TODO: Should we support sparse vectors?
+        String[] out = new String[dimension];
+        int idx = externalVal.indexOf(',');
+        int end = idx;
+        int start = 0;
+        int i = 0;
+        if (idx == -1 && dimension == 1 && externalVal.length() > 0) {//we have a single point, dimension better be 1
+            out[0] = externalVal.trim();
+            i = 1;
+        } else if (idx > 0) {//if it is zero, that is an error
+            //Parse out a comma separated list of values, as in: 73.5,89.2,7773.4
+            for (; i < dimension; i++) {
+                while (start < end && externalVal.charAt(start) == ' ') start++;
+                while (end > start && externalVal.charAt(end - 1) == ' ') end--;
+                if (start == end) {
+                    break;
+                }
+                out[i] = externalVal.substring(start, end);
+                start = idx + 1;
+                end = externalVal.indexOf(',', start);
```

Comment:
Numbers will always be stored as decimals with `.` and separator as `,`? Need to make sure this doesn't mess up in other locales that use comma as the decimal separator.

Reply:
I've added a separator init param and parse doubles using NumberFormat with default locale